core.cc

```diff
@@ -30,7 +30,7 @@ class Worker: public Nan::AsyncWorker {
         void HandleOKCallback() {
             Nan::HandleScope scope;
 
-            if(!callback->IsEmpty())
+            if(callback->GetFunction()->IsFunction())
                 callback->Call(0, NULL);
         };
```

Comment:
Yes, but that is why you should check the result of `IsFunction()` before creating the `Nan::Callback` instance.  Blindly casting the function argument is sliding into UB territory.